torch/testing/_internal/common_utils.py

```diff
@@ -1128,6 +1128,28 @@ def genSparseTensor(self, size, sparse_dim, nnz, is_uncoalesced, device, dtype):
     def safeToDense(self, t):
         return t.coalesce().to_dense()
 
+    # Compares torch function with reference function for given sample input (object of SampleInput)
+    # Note: only values are compared, type comparison is not done here
+    def compare_with_reference(self, torch_fn, ref_fn, sample_input, **kwargs):
+        # Note: if .numpy() fails for some reason, we won't know if it failed for inp/args/kwargs (TODO)
+        n_inp, n_args, n_kwargs = sample_input.numpy()
+        t_inp, t_args, t_kwargs = sample_input.input, sample_input.args, sample_input.kwargs
+
+        actual = torch_fn(t_inp, *t_args, **t_kwargs)
+        expected = ref_fn(n_inp, *n_args, **n_kwargs)
+
```

Comment:
I think this can just do `self.assertEqual(actual, expected, exact_device=False)`

Reply:
Makes sense, passes all the tests on local server. Thanks!